### ISSUE TITLE
When a file change is detected retry loading it when locked

### ISSUE THREAD

**Issue by lygstate** [atom/atom#7659]
```
Unable to read file after file change event. Make sure you have permission to access C:\CI-Cor\ks\bolt\build\ConfigStatusBolt.py.
EBUSY: resource busy or locked, open 'C:\CI-Cor\ks\bolt\build\ConfigStatusBolt.py'

```

**Comment by izuzak** [comment]
@lygstate Can you provide a bit [more information](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#submitting-issues)?

**Comment by blah238** [comment]
@lygstate @izuzak  Pretty sure this is a Windows-only issue. This occurs when saving a file in another program that is currently open in Atom, but is not the active tab. However I have only noticed this with a certain program (TortoiseGitMerge), not general text editors like Notepad.

The steps to reproduce for me are as follows:

1. Do a diff in a repository with staged changes.
2. Open one of the modified files in Atom
3. Open some other file with Atom so the file in step 2 is not the active tab
4. Double-click the file in step 2 in the diff in TortoiseGit
5. Make some changes and save the file

Error above occurs. It would be good to get some more repro cases, if anyone else is experiencing this.

**Comment by lygstate** [comment]
@blah238 Yeap, you are definitely correct, I am facing the same problem with TortoiseGitMerge

**Comment by LePhil** [comment]
I can confirm this in Windows 7 with Atom 1.5.4.
I have a project open in Atom, then git-pull via Tortoise and Atom is "unable to read file after **change** event."

**Comment by Zireael07** [comment]
Does this happen with other Git clients? I'm using SourceTree and haven't noticed anything like that..

**Comment by khanfx** [comment]
This also happens when having a .js file open that is a build target for the Typescript tsc.exe compiler. My suspicion is that it results from an app that holds a file exclusively locked for a short period of time, rather than just doing a near-instantaneous write/replace of the file contents. 

The solution might be an extension, not a core update - one that watches and waits for the file to unlock.

**Comment by victorpolerca** [comment]
I also confirm this using Atom version 1.6.2 on windows 7.
I have the file open in atom, when i update via Tortoise svn atom is unable to read file.

**Comment by jkvierra** [comment]
Was there a solution to this? Currently experiencing this issue.

**Comment by benoitcerrina** [comment]
seeing the same thing when saving a file from the chrome debugger.  It happens even if the file in question is the current active tab.
bypass is to close and reopen the file.

**Comment by EugeneRecto** [comment]
Does anyone has a solution for this? I am experiencing the same problem.

**Comment by kamifun** [comment]
same

**Comment by maxkoryukov** [comment]
> opened this issue on **Jul 3, 2015**

it is **January 14, 2017** already :wink:

**Comment by alancliddell** [comment]
Not just a Windows issue. Was experiencing this with Atom 1.12.7 on Ubuntu 16.10, but seems to be fixed in 1.13.1.

**Comment by manish2535** [comment]
+1

**Comment by pegues** [comment]
I've always had this problem with koala-app. I'd switch over to prepros, but prepros crashes my computer. Very unfortunate that this issue still isn't fixed after experiencing the aggravating EBUSY notice for so long. Perhaps the best way to compile preprocessor files is to do it using Ruby with sass installed.

**Comment by maxkoryukov** [comment]
Probably, PULL REQUESTS are welcome)))

all we need - just fix and send  a patch

**Comment by ceciliayang213** [comment]
I experienced a similar problem with windows 8, Atom 1.16.0, and Source Tree 2.0.19.1: 
Unable to read file after file resurrect event. Make sure you have permission to access C:\Users\USER\Documents\housekeeper\src\models\customers.js
EPERM: operation not permitted, open 'C:\Users\USER\Documents\housekeeper\src\models\customers.js'

This error occurred when I tried to finish a feature.
After reading the discussion here, I tried to close the file (the feature I want to finish) first, 
then tried to finish feature again.
And it worked.
